lib/config/config-cache.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @fileoverview Responsible for caching config files
+ * @author Sylvan Mably
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Constants
+//------------------------------------------------------------------------------
+
+const VECTOR_SEP = ",";
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Get a string hash for a single object from a config vector
+ * @param {Object} entry config vector entry
+ * @returns {string} hash of the vector entry
+ * @private
+ */
+function hashVectorEntry(entry) {
+    return [].concat(entry.filePath, entry.matchingOverrides).join(VECTOR_SEP);
```

Comment:
I don't think we could use a symbol here, because we're essentially doing a deep equality check on the two vectors. We'd have to figure out the correct mapping from a vector to a symbol, which is really the same problem as directly testing the equality of the vectors.